ServerMonitoringSystem/Services/MonitoringService.cs

```diff
@@ -0,0 +1,32 @@
+﻿using ServerMonitoringSystem.Configuration;
+using MessagingLibrary.Interfaces;
+using ServerMonitoringSystem.Domain;
+
+namespace ServerMonitoringSystem.Services;
+
+public class MonitoringService : IMonitoringService
+{
+    private readonly StatisticsCollector _collector;
+    private readonly IMessagePublisher _publisher;
+    private readonly ServerStatisticsConfig _config;
+
+    public MonitoringService(StatisticsCollector collector, IMessagePublisher publisher, ServerStatisticsConfig config)
```

Comment:
use `IStatisticsCollector` instead, so the service can be mocked for unit testing.